File: Templates/Java/src/main/java/ElasticSearch/Hit.java
package ElasticSearch;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Map;
import java.util.Collection;

public class Hit<TDocument> {
    @JsonProperty("fields")
    private Map<String, Object> fields;

    @JsonProperty("highlight")
    private Map<String, Collection<String>> highlight;

    @JsonProperty("_id")
    private String id;

    @JsonProperty("_ignored")
    private Collection<String> ignored;

    @JsonProperty("_index")
    private String index;

    @JsonProperty("matched_queries")
    private Collection<String> matchedQueries;

    @JsonProperty("_node")
    private String node;

    @JsonProperty("_primary_term")
    private Long primaryTerm;

    @JsonProperty("_routing")
    private String routing;

    @JsonProperty("_score")
    private Double score;

    @JsonProperty("_seq_no")
    private Long seqNo;

    @JsonProperty("_shard")
    private String shard;

    @JsonProperty("_source")
    private TDocument source;

    @JsonProperty("_version")
    private Long version;

    // Getters and Setters
    // ...
}

File: Templates/TypeScript/ElasticSearch/hit.ts
import { JsonProperty } from "json-object-mapper";

export class Hit<TDocument> {
	@JsonProperty({ name: "_id", required: true })
	public id: string;

	@JsonProperty({ name: "_index", required: true })
	public index: string;

	@JsonProperty({ required: false })
	public fields?: { [key: string]: any };

	@JsonProperty({ required: false })
	public highlight?: { [key: string]: string[] };

	@JsonProperty({ name: "_ignored", required: false })
	public ignored?: string[];

	@JsonProperty()
	public matchedQueries?: string[];

	@JsonProperty({ name: "_node", required: false })
	public node?: string;

	@JsonProperty({ name: "_primary_term", required: false })
	public primaryTerm?: number;

	@JsonProperty({ name: "_routing", required: false })
	public routing?: string;

	@JsonProperty({ name: "_score", required: false })
	public score?: number;

	@JsonProperty({ name: "_seq_no", required: false })
	public seqNo?: number;

	@JsonProperty({ name: "_shard", required: false })
	public shard?: string;

	@JsonProperty({ name: "_source", required: false })
	public source?: TDocument;

	@JsonProperty({ name: "_version", required: false })
	public version?: number;
}
